From b1855f751a4b9f9a0a55ae8cc49b4414fca0b34d Mon Sep 17 00:00:00 2001
From: Tom Rondeau <tom@trondeau.com>
Date: Thu, 10 Apr 2014 15:18:30 -0400
Subject: [PATCH 15/15] digital: fix for issue #664.

The mpsk_receiver_cc block needed to publicly inherit from control_loop.
---
 gr-digital/include/gnuradio/digital/mpsk_receiver_cc.h |    5 ++++-
 gr-digital/lib/mpsk_receiver_cc_impl.h                 |    4 +---
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/gr-digital/include/gnuradio/digital/mpsk_receiver_cc.h b/gr-digital/include/gnuradio/digital/mpsk_receiver_cc.h
index 94e765d..df67609 100644
--- a/gr-digital/include/gnuradio/digital/mpsk_receiver_cc.h
+++ b/gr-digital/include/gnuradio/digital/mpsk_receiver_cc.h
@@ -25,6 +25,7 @@
 
 #include <gnuradio/digital/api.h>
 #include <gnuradio/block.h>
+#include <gnuradio/blocks/control_loop.h>
 
 namespace gr {
   namespace digital {
@@ -65,7 +66,9 @@ namespace gr {
      * self-noise.
      *
      */
-    class DIGITAL_API mpsk_receiver_cc : virtual public block
+    class DIGITAL_API mpsk_receiver_cc
+      : virtual public block,
+        virtual public blocks::control_loop
     {
     public:
       // gr::digital::mpsk_receiver_cc::sptr
diff --git a/gr-digital/lib/mpsk_receiver_cc_impl.h b/gr-digital/lib/mpsk_receiver_cc_impl.h
index f17f88c..8e404da 100644
--- a/gr-digital/lib/mpsk_receiver_cc_impl.h
+++ b/gr-digital/lib/mpsk_receiver_cc_impl.h
@@ -25,7 +25,6 @@
 
 #include <gnuradio/digital/mpsk_receiver_cc.h>
 #include <gnuradio/attributes.h>
-#include <gnuradio/blocks/control_loop.h>
 #include <gnuradio/gr_complex.h>
 #include <fstream>
 #include <gnuradio/filter/mmse_fir_interpolator_cc.h>
@@ -33,8 +32,7 @@
 namespace gr {
   namespace digital {
 
-    class mpsk_receiver_cc_impl
-      : public mpsk_receiver_cc, public blocks::control_loop
+    class mpsk_receiver_cc_impl : public mpsk_receiver_cc
     {
     public:
       mpsk_receiver_cc_impl(unsigned int M, float theta, 
-- 
1.7.10.4

